rcl/src/rcl/init.c

```diff
@@ -67,6 +67,7 @@ rcl_init(
   rcl_allocator_t allocator = options->impl->allocator;
   RCL_CHECK_ALLOCATOR(&allocator, return RCL_RET_INVALID_ARGUMENT);
   RCL_CHECK_ARGUMENT_FOR_NULL(context, RCL_RET_INVALID_ARGUMENT);
+  RCUTILS_LOGGING_AUTOINIT_WITH_ALLOCATOR(allocator);
```

Comment:
This is actually the fix to take user defined allocator in account.

Reply:
I think adding this is enough in `rcl`, and then fix the `rcutils` by updating the default log level with an environment variable if exists.